pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,373 @@
+	if err != nil {
+		log.Errorf("error closing stat file handle %v", err)
+	}
+	return err
+}
+
+// SetupFlowHandle generates a windows Driver Handle, and creates a DriverHandle struct to pull flows from the driver
+// by setting the necessary filters
+func (di *DriverInterface) SetupFlowHandle() error {
+	h, err := di.GenerateDriverHandle()
+	if err != nil {
+		return err
+	}
+	dh, err := NewDriverHandle(h, FlowHandle)
+	if err != nil {
+		return err
+	}
+
+	// Prepare handle filters for each interface
+	err = dh.createDriverHandleFilters()
+	if err != nil {
+		return err
+	}
+
+	// Set handle filters
+	err = dh.setFilters()
+	if err != nil {
+		return err
+	}
+
+	di.driverFlowHandle = dh
+	return nil
+}
+
+// SetupStatsHandle generates a windows Driver Handle, and creates a DriverHandle struct
+func (di *DriverInterface) SetupStatsHandle() error {
+	h, err := di.GenerateDriverHandle()
+	if err != nil {
+		return err
+	}
+
+	dh, err := NewDriverHandle(h, StatsHandle)
+	if err != nil {
+		return err
+	}
+
+	di.driverStatsHandle = dh
+	return nil
+
+}
+
+// GenerateDriverHandle creates a new windows handle attached to the driver
+func (di *DriverInterface) GenerateDriverHandle() (windows.Handle, error) {
+	p, err := windows.UTF16PtrFromString(di.path)
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Debug("Creating Driver handle...")
+	h, err := windows.CreateFile(p,
+		windows.GENERIC_READ|windows.GENERIC_WRITE,
+		windows.FILE_SHARE_READ|windows.FILE_SHARE_WRITE,
+		nil,
+		windows.OPEN_EXISTING,
+		0,
+		windows.Handle(0))
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Info("Connected to driver and handle created")
+	return h, nil
+}
+
+// CloseDriverHandle closes an open handle on the driver
+func (di *DriverInterface) CloseDriverHandle(handle windows.Handle) error {
+	return windows.CloseHandle(handle)
+}
+
+func (di *DriverInterface) getStats() (map[string]interface{}, error) {
+
+	flowHandleStats, err := di.driverFlowHandle.getStatsForHandle()
+	if err != nil {
+		return nil, err
+	}
+
+	totalDriverStats, err := di.driverStatsHandle.getStatsForHandle()
+	if err != nil {
+		return nil, err
+	}
+	totalFlows := atomic.LoadInt64(&di.totalFlows)
+
+	return map[string]interface{}{
+		"driver_total_flow_stats":  totalDriverStats,
+		"driver_flow_handle_stats": flowHandleStats,
+		"total_flows": map[string]int64{
+			"total": totalFlows,
+		},
+	}, nil
+}
+
+func (di *DriverInterface) getFlows(waitgroup *sync.WaitGroup) (flows []*C.struct__perFlowData, error error) {
+	waitgroup.Add(1)
+	defer waitgroup.Done()
+
+	readbuffer := make([]uint8, 1024)
+
+	for {
+		var count uint32
+		bytesused := int(0)
+		err := windows.ReadFile(di.driverFlowHandle.handle, readbuffer, &count, nil)
+		if err != nil {
+			if err != windows.ERROR_MORE_DATA {
+				log.Info(err)
+				break
+			}
+		}
+		var buf []byte
+		for ; bytesused < int(count); bytesused += C.sizeof_struct__perFlowData {
+			buf = readbuffer[bytesused:]
```

Comment:
I think you'll run into issues when the outermost for-loop goes around more than once because you'll overwrite `readbuffer`

Reply:
Fixing this in:

- https://github.com/DataDog/datadog-agent/pull/5388/files

resolving this.